helix-term/src/ui/markdown.rs

```diff
@@ -164,22 +171,51 @@ impl Markdown {
             .map(|key| get_theme(key))
             .collect();
 
-        let mut list_stack = Vec::new();
-
         for event in parser {
             match event {
-                Event::Start(Tag::List(list)) => list_stack.push(list),
+                Event::Start(Tag::List(list)) => {
+                    // if the list stack is not empty this is a sub list, in that
+                    // case we need to push the current line before proceeding
+                    if !list_stack.is_empty() {
+                        push_spans(&mut spans, &mut lines);
+                    }
+
+                    list_stack.push(list);
+                }
                 Event::End(Tag::List(_)) => {
                     list_stack.pop();
-                    // whenever list closes, new line
-                    lines.push(Spans::default());
+
+                    // whenever top-level list closes, empty line
+                    if list_stack.is_empty() {
+                        lines.push(Spans::default());
+                    }
                 }
                 Event::Start(Tag::Item) => {
+                    // FIXME: What to do in this case? This is a bug in the markdown parser and we
+                    // cannot really continue, but obviously we shouldn't crash the editor.
+                    assert!(list_stack.len() > 0, "Cannot start an item without a list");
```

Comment:
This will have to be fixed before this can be merged. If this assert triggers there is a bug in the markdown parsing, but the editor shouldn't crash because of it.